Assets/MixedRealityToolkit/_Core/Services/BaseDataProvider.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Core.Interfaces;
+
+namespace Microsoft.MixedReality.Toolkit.Core.Services
+{
+    /// <summary>
+    /// The base data provider implements <see cref="IMixedRealityDataProvider"/> and provides default properties for all data providers.
+    /// </summary>
+    /// <remarks>
+    /// Empty, but reserved for future use, in case additional <see cref="IMixedRealityDataProvider"/> properties or methods are assigned.
+    /// </remarks>
+    public abstract class BaseDataProvider : BaseServiceWithConstructor, Interfaces.IMixedRealityDataProvider
```

Comment:
Curious if we actually need a contstructor-less base implementation. Wouldn't having a base and a basewithconstructor create potential traps (read: easy to make mistakes) for customers creating services and providers?